fe/fe-core/src/main/java/com/starrocks/analysis/ColumnDef.java

```diff
@@ -86,7 +86,11 @@ public DefaultValueDef(boolean isSet, Expr expr) {
     private final TypeDef typeDef;
     private AggregateType aggregateType;
     private boolean isKey;
-    private boolean isAllowNull;
+    // For primary key column null constraint, when null specify, should report an error, when not null specify, should working as normal, when implicit specified, default behavior should be working as not null specify.
+    // For now, just two value assign to variable `isAllowNull`, a two state boolean value named isAllowNull to hold null constraint value, which can not tell the difference between implicit specified and explicit specify null, so we can not make a decision, to working as normal, or report an error, that is the question.
```

Comment:
Now, `isAllowNull` is used to indicate a null constraint hold or not. Primary-key and non-primary-key columns obey different constraints, so the `isAllowNull` can not be assigned a default value.
Add a new variable name `isAllowNullImplicit` to indicate the message. If`isAllowNullImplicit=true`, it indicates the null constraint is obeyed implicitly.